pkg/chartutil/create.go

```diff
@@ -189,6 +190,10 @@ spec:
         app.kubernetes.io/name: {{ include "<CHARTNAME>.name" . }}
         app.kubernetes.io/instance: {{ .Release.Name }}
     spec:
+      {{- if .Values.image.pullSecret -}}
```

Comment:
`imagePullSecrets` is a list. I'd suggest to respect that so it is possible to actually configure a list.

I would not nest it under `image` since it is not related to that but a field that's part of the pod spec. This will become obvious as soon as you have multiple containers in the pod.

Here's an example:
* https://github.com/helm/charts/blob/master/stable/mongodb-replicaset/templates/mongodb-statefulset.yaml#L43-L47
* https://github.com/helm/charts/blob/master/stable/mongodb-replicaset/values.yaml#L25

Reply:
And to be consistent, please use `with`.